x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -435,10 +435,16 @@ else if (plan instanceof OrderBy) {
                 }
             }
 
-            return result;
+            return filterUnsupportedProjections(result);
```

Comment:
Why this change?

Reply:
This is needed since, without it, if an index has an `unsigned_long` field and a pre-unsigned_long client (ex 7.7.0) executes a `SELECT * FROM ...` the "expanded" set of fields will contain this UL field that will get caught in the verifier (`#checkClientSupportsDataTypes()`), which checks if a query from an old client tries to select "new" fields that the client would not understand.